### PR TITLE
Adding T_from_p_h into SinglePhaseFluidProperties

### LINKED ISSUE #12108: Add T_from_p_h() into SinglePhaseFluidProperties interface
## Rationale

SAM needs this fluid property call to compute temperature at a branch when `p` and `h` is known. This would be also useful to codes that solves in terms of p and h (like FALCON)

## Description

Add a new fluid property lookup API for temperature when pressure and enthalpy is known. The method names will be `T_from_p_h`.

## Impact

None, adding new interface.

### PR DESCRIPTION
**Pull request by andrsd**
Included a unit test.

Closes #12108

### CODE REVIEW

File: modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C [hunk 1]
```diff
@@ -491,3 +491,8 @@ Real SinglePhaseFluidProperties::g_from_v_e(Real, Real) const
 {
   mooseError(name(), ": ", __PRETTY_FUNCTION__, " not implemented.");
 }
+
+Real SinglePhaseFluidProperties::T_from_p_h(Real, Real) const
+{
+  mooseError(name(), ": ", __PRETTY_FUNCTION__, " not implemented.");
```
Comment (joshuahansel) [inline modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: There should be a default implementation here - there is a path to T from p and h:
```
s = s_from_h_p(h,p)
rho = rho_from_p_s(p,s)
v = 1/rho
e = e_from_v_h(v,h)
T = T_from_v_e(v,e)
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/fluid_properties/include/userobjects/SinglePhaseFluidProperties.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ class SinglePhaseFluidProperties : public FluidProperties
    *
    * @param[in] v   specific volume
    * @param[in] e   specific internal energy
-   * @returns sound speed
+   * @return sound speed
```
Comment (joshuahansel) [inline modules/fluid_properties/include/userobjects/SinglePhaseFluidProperties.h]: If you just want consistent style with the rest of the file, that's fine, but I think `@return` and `@returns` are both valid Doxygen commands.

Comment (andrsd) [reply modules/fluid_properties/include/userobjects/SinglePhaseFluidProperties.h]: You are right. I thought they only had `@return`

File: modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C (hunk 1 above)
Comment (andrsd) [reply modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: This is true. But,  for this implementation people would need to have `s_from_h_p` and `rho_from_p_s`. Now, is it simpler to implement these 2 methods or just this one directly if they needed it?

Comment (joshuahansel) [reply modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: If people don't have `s_from_h_p` and `rho_from_p_s`, I think the default implementation is still fine - they'll just get the error messages corresponding to those calls. If there is a more direct implementation of `T_from_p_h` for a specific fluid properties class, then great, implement that one and don't worry about `s_from_h_p` and `rho_from_p_s`. But that doesn't mean that the default implementation can't still be what I outlined above.

Comment (andrsd) [reply modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: That was kind of my point. If there is a direct implementation, there is no need for the default one. If there is not, then you need `s_from_h_p` _and_ `rho_from_p_s`, which IMO is more difficult to get. But the default implementation is not gonna hurt anything, I guess...

Comment (joshuahansel) [reply modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: Right, the purpose of the default implementation is for fluid properties that don't have a direct implementation for `T_from_h_p`, but do have `s_from_h_p` and `rho_from_p_s`.

Comment (andrsd) [reply modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C]: I will make the change

**Review by joshuahansel — APPROVED** [review]

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**